### DISCUSSION TITLE
Door and Doorframe sprite depth inconsistent with other sprite depths

### DISCUSSION THREAD

**Discussion by nomnivore1** [FakeFishGames/Barotrauma#14552] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Working in the editor, I'm trying to layer doors so that the doorframe is drawn _in front of_ wires (sprite depth of 0.8) but _behind_ players. This seems impossible, and it means that any wires added in gameplay will connect to the center of the doorway, creating a visible wire that nobody can get rid of. I'm trying to avoid this. 

Setting the sprite depth in the editor controls the depth of the door sprite itself, and the doorframe sprite seems to move with this. Setting the depth of the door to 0.5 causes the doorframe to be drawn above wires. Hooray! but the doorframe now draws above players as well. 

In short, changing the depth of the *door* sprite from 0.51 to 0.5 causes the *doorframe* sprite's depth to jump from >0.8 (drawn behind wires) to <0.5 (drawn over players) making it impossible to draw a doorframe somewhere between players and wires. 

![image](https://github.com/user-attachments/assets/b0d1c47e-c5ff-4f8e-a6a5-2b2856237912)
![image2](https://github.com/user-attachments/assets/19869899-0965-4b10-b910-303e3d8c4e95)

image 1: door sprite set to 0.51, doorframe draws behind players (0.5) and behind wires (0.8)
image 2: door sprite set to 0.5, draws in front of players and wires. 

If the doorframe sprite is set to the depth of the door +0.3, this is a bug and it should not be drawing above players. 
if the doorframe sprite has two depth values that it selects between depending on the depth of the door sprite, this prevents wires running behind doors without drawing the doors above the players, making clean wiring in game impossible. 

### Reproduction steps

1. place a door in the editor, set it to open, run a wire through the doorway. 
2. test and observe that, using the default sprite depths, the doorframe is behind the player and the wire (must have a depth <0.8)
3. set the door sprite depth from 0.51 to 0.5
4. test and observe that the doorframe is now in front of both the player and the wire

a change of 0.51 to 0.5 in the door sprite depth has changed the doorframe sprite depth from >0.8 to <0.5

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by moleeeeee** [comment]
I know this is a bit late, but the only way is to hide the wires into the lower/front sprite depth like your items and walls/flooring if you decide to raise the sprite depth on the wallpapers. However, you can always selection the option to hide the wires in-game, leaving the wallpaper depth alone at its default depth 980 or whatever it is.
